core/util/src/main/scala/net/liftweb/util/CurrencyZone.scala

```diff
@@ -32,29 +32,29 @@ trait DollarCurrency extends TwoFractionDigits {
 /* Various Currencies */
 object AU extends CurrencyZone {
     type Currency = AUD
-    var locale = new Locale("en", "AU")
-    def make(x: BigDecimal) = new Currency{def amount = x}
+    var locale: Locale = Locale.of("en", "AU")
```

Comment:
Locale.of was introduced in java 19. This line isn't kosher for 4.0.

Reply:
I tested merging this into 5.0, since there were also warnings fixed there. The merge was clean but I get an error on TimeHelpersSpec that I haven't been able to track down.

Reply:
Ah, good catch ! I tried to find which one was OK on JDK 11 and... Well that was not a success then :) So I'm not sure we have something working on 11 to 21 without deprecation warning

Reply:
For the only change in `TimeHelpersSpec` is the new path for collection converter. It works on my branch (with JDK 19 if it matters)

Reply:
> Ah, good catch ! I tried to find which one was OK on JDK 11 and... Well that was not a success then :) So I'm not sure we have something working on 11 to 21 without deprecation warning

Yeah I think we just deal with the deprecation warning in this case! :sob:

Reply:
I verified that TimeHelpersSpec is failing for me in lift-5.0 before this PR was merged too, so I'm still not sure what's causing it, but it's not this PR.